src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash -eu
+#
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Unit tests for tools/test/collect_cc_code_coverage.sh
+
+
+# Load the test setup defined in the parent directory
+CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${CURRENT_DIR}/../integration_test_setup.sh" \
+  || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
+
+
+function setup_cc_sources() {
+  mkdir coverage
+  cat << EOF > coverage/a.h
+int a(bool what);
+EOF
+
+  cat << EOF > coverage/a.cc
+#include "a.h"
+
+int a(bool what) {
+  if (what) {
+    return 1;
+  } else {
+    return 2;
+  }
+}
+EOF
+
+  cat << EOF > coverage/t.cc
+#include <stdio.h>
+#include "a.h"
+
+int main(void) {
+  a(true);
+}
+EOF
+}
+
+function setup_gcc_gcda_files() {
+  cd coverage/
+  # Generate .gcno files.
+  g++ -fprofile-arcs -ftest-coverage -c a.h a.cc t.cc
+  # Produce instrumented binary.
+  g++ -fprofile-arcs -ftest-coverage a.h a.cc t.cc -o test
+  # Generate test.gcda file.
+  ./test
+  cd ..
+}
+
+function setup_script_environment() {
+  export COVERAGE_DIR="$PWD/coverage"
+  export ROOT="$PWD"
+  export COVERAGE_OUTPUT_FILE="$PWD/coverage_report.dat"
+  export COVERAGE_MANIFEST="$PWD/coverage_manifest.txt"
+
+  # The script expects gcov to be at $COVERAGE_DIR/gcov.
```

Comment:
You should explain why the script expects those and if there's no explanation then state here that that's a "bug" you're trying to fix (and how).

Reply:
I'm not sure why should it be explained in the test? There is already an explanation in the actual script.